### DISCUSSION TITLE
PubSub resubscribing

### DISCUSSION THREAD

**Discussion by Swedz** [xetorthio/jedis#2334] [Q&A] [answered]
Is there an effective way of re-subscribing in order to add/remove channels that you would be subscribed to? Should I call `JedisPubSub#unsubscribe(String...)`? Can this method be called from a separate thread than the pubsub is currently running on?

**Comment by sazzad16** [comment, marked as answer]
@Swedz 

> Is there an effective way of re-subscribing in order to add/remove channels that you would be subscribed to?

unsubscribe and subscribe is the way to go

> Can this method be called from a separate thread than the pubsub is currently running on?

yes
